### ISSUE TITLE
[1300.0.7] Bots still cant handle empty oxy masks

### ISSUE THREAD

**Issue by CurePindal** [Regalis11/Barotrauma#5471]
- [x ] I have searched the issue tracker to check if the issue has already been reported.

**Description**
Docked at an abandoned station, my subs reactor was turned off, room oxy depleted and bots started to use their masks. (see pic below)
When the mask oxy ran dry they didnt replace it. Spare oxy tanks where available in containers no 3m away.
When room oxy was restored, bots didnt take off their masks.
Most bots where set to hold pos command, but even without they didnt handle their masks and/or empty oxy tanks.

Expected behaviour:
- bots replace oxy tanks in their masks when needed
- bots stick empty oxy tanks into the oxy gen - its even in the same room and has no tanks in (ok, might default to the cleanup job - havent tested this)
- bots take off oxy masks when room oxy gets restored
- would be lovely: when they take their masks off, bots restore their empty tanks in the masks with fresh tanks. (proper prep saves lifes)

**Steps To Reproduce**
Dock ship at (in this case) an abandoned outpost
Turn off the reactor
Bots start to use the oxy masks that they carry once room oxy runs dry.

then: Restore room oxy, disable all commands on a bot and he stays with the mask on.

**Version**
Win 10, unstable vanilla, SP

**Additional information**
![20210408155446_1](https://user-images.githubusercontent.com/3466193/114041443-d9ac8800-9884-11eb-99ee-393863d9d677.jpg)

**Comment by itchyOwl** [comment]
The bots don't unequip diving gear when they are not in a friendly sub or when they are waiting, which is intentional. But I think it shouldn't apply to masks and I changed it (Regalis11/Barotrauma-development@727f843).

> When the mask oxy ran dry they didnt replace it. Spare oxy tanks where available in containers no 3m away.

I found one issue where the bots fail to run for the safety when they can't find any oxygen, but I'm not sure it's causing this. This issue is fixed in Regalis11/Barotrauma-development@0d6bef0.

Generally the bots seemed to replace tanks when they should and seek diving gear when needed. But it's possible that there are corner cases where they fail to do that. For such corner cases, the steps to repro need to be very specific, or it's difficult to fix it.

The logic for all the possible cases where the bots need the mask or the suit and when they should put them in place or just unequip them is surprisingly complex. And there's been a lot of issues related to it. So I wouldn't be surprised if there's still some things wrong in it. But in general I think it works now much better than for example a year ago. The progress couldn't be done without you guys constantly testing and writing issues when something is not as it should. So keep it going, and thanks for the report!
